subprojects/docs/src/docs/userguide/basePlugin.adoc

```diff
@@ -0,0 +1,90 @@
+== The Base Plugin
+
+The Base Plugin provides features that are common to most builds and adds a structure to the build that promotes consistency in how they are run. Its most significant contribution is a set of <<sec:base_tasks,'lifecycle' tasks>> that act as an umbrella for the more specific tasks provided by other plugins and build authors.
+
+[[sec:base_plugin_usage]]
+=== Usage
+
+----
+plugins {
+    id 'base'
+}
+----
+
+[[sec:base_tasks]]
+=== Tasks
+
+`clean` — type: `Delete`::
+Deletes the build directory and everything in it, i.e. the path specified by the `buildDir` project property.
+
+`check` — _lifecycle task_::
+Plugins and build authors should attach their verification tasks, such as ones that run tests, to this lifecycle task using `check.dependsOn(__task__)`.
+
+`assemble` — _lifecycle task_::
+Plugins and build authors should attach tasks that produce distributions and other consumable artifacts to this lifecycle task. For example, `jar` produces the consumable artifact for Java libraries. Attach tasks to this lifecycle task using `assemble.dependsOn(__task__)`.
+
+`build` — _lifecycle task_::
+(Depends on `assemble` and `check`) Builds everything
+
+`build__Configuration__` — build rule::
+Assembles those artifacts attached to the named configuration. For example, `buildArchives` will execute any task that is required to create any artifact attached to the `archives` configuration.
+
+`upload__Configuration__` — build rule::
+Does the same as `build__Configuration__`, but also uploads all the artifacts attached to the given configuration.
+
+`clean__Task__` — build rule::
+Removes the <<sec:task_inputs_outputs,defined outputs>> of a task, e.g. `cleanJar` will delete the JAR file produced by the `jar` task of the Java Plugin.
+
+
+[[sec:base_plugin_configurations]]
+=== Dependency management
+
+The Base Plugin adds no <<managing_dependency_configurations,configurations for dependencies>>, but it does add the following configurations for <<sec:artifacts_and_configurations,artifacts>>:
+
+`default`::
+By default, contains all the artifacts attached to all other configurations. It's used by consumer projects. For example, if project B has a `runtimeOnly` dependency on project A, then the artifacts in project A's `default` configuration are added to project B's `runtimeOnly` configuration.
+
+`archives`::
+A standard configuration for the production artifacts of a project. This results in an `uploadArchives` task for publishing artifacts attached to the `archives` configuration.
+
+Note that the `assemble` task generates all artifacts that are attached to the `archives` configuration.
+
+[[sec:base_plugin_conventions]]
+=== Conventions
+
+The Base Plugin only adds conventions related to the creation of archives, such as ZIPs, TARs and JARs. Specifically, it provides the following project properties that you can set:
+
+`archivesBaseName` — default: `$project.name`::
+Provides the default `baseName` for archive tasks.
```

Comment:
Can we add the API link to `baseName` here?

Reply:
Sure. Again, it will look ugly. Hopefully that can be fixed at some point.

- [ ] Link to API docs for `baseName`

Reply:
@bmuschko `AbstractArchiveTask` doesn't appear to be in the DSL reference. Should it be?

Reply:
My mistake. It is, but it's not in the left-hand nav bar for some reason.

Reply:
No problem. I think the left hand nav bar only lists selected classes.